### PR TITLE
Fetch/XHR: test JSON parser

### PR DESCRIPTION
**Pull request by annevk**
Rather unfortunate that Chrome and Safari (still) seem to support UTF-16 as well here...

### CODE REVIEW

**Comment by annevk** [comment]
Filed these bugs:

* https://bugs.chromium.org/p/chromium/issues/detail?id=1189627
* https://bugs.webkit.org/show_bug.cgi?id=223441

**Comment by yutakahirano** [comment]
With `fetch("/xhr/resources/utf16-bom.json").then(r => r.json())` on wpt.live Chrome outputs

```
Uncaught (in promise) SyntaxError: Unexpected token � in JSON at position 0
```

, so it seems not supported.

**Comment by yutakahirano** [comment]
Ahh, ok, not supported in fetch but supported on XHR.

File: fetch/api/response/json.any.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  const response = await fetch(`data:,\uFEFF{ "b": 1, "a": 2, "b": 3 }`);
+  const json = await response.json();
+  assert_array_equals(Object.keys(json), ["b", "a"]);
+  assert_equals(json.a, 2);
+  assert_equals(json.b, 3);
+}, "Ensure the correct JSON parser is used");
+
+promise_test(async t => {
+  const response = await fetch("/xhr/resources/utf16-bom.json");
+  return promise_rejects_js(t, TypeError, response.json());
```
Comment (yutakahirano) [inline fetch/api/response/json.any.js:13]: SyntaxError?

Comment (annevk) [reply fetch/api/response/json.any.js]: Thanks! I must have mixed up URLs locally as I thought this was passing in Firefox and failing in Chrome, but I guess it was failing in both as now they both pass...

**Review by yutakahirano — APPROVED** [review]
LGTM. I'll try to measure the impact and remove the support on Chromium for the XHR incompatibility.

**Pull request merged**